rolling-shutter/medley/broker.go

```diff
@@ -0,0 +1,184 @@
+package medley
+
+import (
+	"sync"
+
+	"github.com/rs/zerolog/log"
+)
+
+// `Broker` allows to distribute a value to multiple
+// receiving channels, so that all channels receive the same value.
+// This is different to the Fan-Out pattern where multiple channels
+// are listening for a send on one channel, but only the first channel
+// to select the receive operation receives this value.
+//
+// The Broker allows to set whether the internal send operation is
+// blocking or non blocking. In the blocking case, all receiving
+// (subscribed) channels have to successively select the receive in order
+// for the broker to continue operation. So if one receiver blocks,
+// everything else will block as well.
+// In the non-blocking case, a subscribed channel that is blocking
+// while the published value is sent will miss the value.
+//
+// One has to be careful in when to select which case.
+// The blocking behavior can be partially subverted by increasing
+// the subscribers buffer size - but if a subscriber is blocked forever,
+// this will only work until this subscribers buffer is full.
+type Broker[T any] struct {
+	mux                  sync.RWMutex
+	stop                 chan struct{}
+	publish              chan T
+	subscribe            chan chan T
+	unsubscribe          chan chan T
+	running              bool
+	bufferSizeSubscriber int
+	blockingSend         bool
```

Comment:
Do we expect to need non-blocking send? It seems dangerous because not only could some receivers not receive all values, but some values might be lost completely. And initializing `NewBroker` with `false` instead of `true` seems like a very easy to miss and hard to find bug. So if we don't have a clear use case already, I'd suggest to simply remove it.

Reply:
To the first question: yes, e.g. in time critical applications where it's more important for all receivers to get a chance at receiving than to have delivery guarantees. Also, you can mitigate slow receivers with buffering the channel - they will be able to still get the value as soon as they are idle.
If one receiver still blocks forever on a channel with a full buffer, at least all other channels can still receive.
It is used in the EpochTicker, although this might change since it still has unfavourable properties (as noted in the source).

The initialisation is a good point. We could negate it and call it `nonBlockingSend` if we think this is the default case. One thing to note, a blocking send can be equally bad if one receiver might block forever, so a sensible choice has to be made anyways.

Reply:
Changed the default to `nonBlockingSend=false`